cumulus/polkadot-parachain/polkadot-parachain-lib/src/command.rs

```diff
@@ -59,11 +56,27 @@ fn new_node_spec(
 
 	Ok(match runtime {
 		Runtime::Shell => Box::new(ShellNode),
-		Runtime::Omni(consensus) => match consensus {
-			Consensus::Aura(AuraConsensusId::Sr25519) =>
-				new_aura_node_spec::<AuraRuntimeApi, AuraId>(extra_args),
-			Consensus::Aura(AuraConsensusId::Ed25519) =>
-				new_aura_node_spec::<AssetHubPolkadotRuntimeApi, AssetHubPolkadotAuraId>(extra_args),
+		Runtime::Omni(block_number, consensus) => match (block_number, consensus) {
```

Comment:
This match seems to grow exponentially in arms if coded as-is. Perhaps not now, but as we get more things to be parameterize-able, we should think about passing arguments down in a less verbose manner.

Reply:
Yes, that's right. I think the only way to split it is to create multiple functions. For the moment I added one more `new_aura_node_spec` fn that takes that aura_id as a parameter, and here we just check the block number. I don't know if this split is super natural. But as we add more parameters, we can play and adjust this split.